### ISSUE TITLE
Unclosed socket after calling the buckets_api

### ISSUE THREAD

**Issue by tmachnitzki** [influxdata/influxdb-client-python#66]
When running Unittest on the following minimal example of code, I get a Warning that not all resources are closed.

```python

import unittest
import os

from influxdb_client import InfluxDBClient
import tracemalloc

tracemalloc.start()


class InfluxTest(unittest.TestCase):

    def setUp(self) -> None:
        self.client = InfluxDBClient(url=os.environ.get("INFLUX_URL"),
                                     token=os.environ.get("INFLUX_TOKEN"))
        self.bucket_api = self.client.buckets_api()

    def tearDown(self) -> None:
        self.client.__del__()
        del self.bucket_api

    def test_connection(self):
        self.bucket_api.find_buckets().to_dict()
```

When I use other APIs like the query_api, I don't get that warning. Does it have something to do with the bucket API not having a `__del__()` method?

This is the warning I get (I replaced my ip with xxxx):

```python
ResourceWarning: unclosed <socket.socket fd=5, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('x.x.x.x', 50040), raddr=('x.x.x.x', 9999)>
  del self.bucket_api
Object allocated at (most recent call last):
  File "/Users/myuser/opt/anaconda3/envs/myenv/lib/python3.7/site-packages/urllib3/util/connection.py", lineno 65
    sock = socket.socket(af, socktype, proto)`

**Comment by bednar** [comment]
Hi @tmachnitzki,

I tried your code by: `pytest github_test.py` and everything works correctly.

My environment is: `platform darwin -- Python 3.7.6, pytest-5.3.5, py-1.8.1, pluggy-0.13.1`

Please, could you provide more info how to simulate it?

Regards

**Comment by tmachnitzki** [comment]
I use `Python 3.7.6` on `macOS Catalina`. Unfortunately I am not able to find out the unittest version number, as it is neither listed in conda or pip, nor does the package has a `__version__` string. 

Using `pytest` I am not getting the warning either, only using `unittest` with
`python -m unittest github_test.py`.

**Comment by bednar** [comment]
The problem is caused by unclosed `pool_manager`: https://github.com/influxdata/influxdb-client-python/blob/ce818996418901e9f768d4b7e656a1e993059a3e/influxdb_client/rest.py#L97

The `pool_manager` should be closed at:

https://github.com/influxdata/influxdb-client-python/blob/b0cb4abc8dadd2d4184a044b0837d4e00419d9b2/influxdb_client/api_client.py#L80

**Comment by bednar** [comment]
Hi @tmachnitzki,

The issue is fixed in [1.5.0 milestone](https://github.com/influxdata/influxdb-client-python/milestone/7). 

If you would like to use a dev version then install client via:

```pip install git+https://github.com/influxdata/influxdb-client-python.git@master```

Regards

**Comment by tmachnitzki** [comment]
I can confirm that it is solved, thanks for the support 😊
